lib/libs/sink-lib.ts

```diff
@@ -90,21 +90,22 @@ const prettyPrintJsonInObject = (obj: any): any => {
 export function getDomainAndNamespace<T extends BaseIndex>(
   baseIndex: T,
 ): { domain: string; index: `${string}${T}` };
-export function getDomainAndNamespace(): { domain: string; index: string };
+
 export function getDomainAndNamespace(baseIndex?: BaseIndex) {
   const domain = process.env.osDomain;
 
   if (domain === undefined) {
     throw new Error("osDomain is undefined in environment variables");
   }
 
-  const indexNamespace = process.env.indexNamespace;
-
-  if (indexNamespace === undefined) {
+  if (indexNamespace !== "" && indexNamespace === undefined) {
     throw new Error("indexName is undefined in environment variables");
   }
 
-  return { index: baseIndex ? `${indexNamespace}${baseIndex}` : indexNamespace, domain };
+  const indexNamespace = process.env.indexNamespace || "";
```

Comment:
Might wanna be careful with `||` since it'll fallback to `""` if `indexNamespace` is an empty string. `??` might be what you're looking for since it'll fallback only for `undefined` and `null`